src/wallet/walletutil.cpp

```diff
@@ -57,23 +57,48 @@ static bool IsBerkeleyBtree(const fs::path& path)
 std::vector<fs::path> ListWalletDir()
 {
     const fs::path wallet_dir = GetWalletDir();
+    const fs::path data_dir = GetDataDir();
+    const fs::path blocks_dir = GetBlocksDir();
+
     const size_t offset = wallet_dir.string().size() + 1;
     std::vector<fs::path> paths;
     boost::system::error_code ec;
 
+    bool skip_path;
+    // Here we place the top level dirs we want to skip in case walletdir is datadir or blocksdir
+    const std::vector<fs::path> ignore_paths = {
+                                        blocks_dir,
+                                        data_dir / "testnet3",
+                                        data_dir / "regtest",
+                                        data_dir / "chainstate",
+                                        data_dir / "database",
+                                        data_dir / "indexes",
```

Comment:
Consider adding "blktree" and "coins" (see `files.md`)

Reply:
Maybe also the default blocks dir too

Reply:
const std::vector<fs::path> ignore_paths = {
                                        blocks_dir,
                                        data_dir / "blktree",